src/sycl/Utils.h

```diff
@@ -206,6 +206,27 @@ int get_min(Func limit_func, int X, Args*... args) {
     }                                                                                                       \
   }
 
+#define PRIVATE_CASE_TYPE_OUTPLACE(enum_type, type, ...) \
+  case enum_type: {                                      \
+    using scalar_t = type;                               \
+    __VA_ARGS__();                                       \
+    break;                                               \
+  }
+
+#define SYCL_DISPATCH_FLOATING_TYPES_AND2(SCALARTYPE1, SCALARTYPE2, TYPE, NAME, ...)    \
+  {                                                                                     \
+    const auto& the_type = TYPE;                                                        \
+    at::ScalarType _st = ::detail::scalar_type(the_type);                               \
+    switch (_st) {                                                                      \
+      PRIVATE_CASE_TYPE_OUTPLACE(at::ScalarType::Double, double, __VA_ARGS__)           \
+      PRIVATE_CASE_TYPE_OUTPLACE(at::ScalarType::Float, float, __VA_ARGS__)             \
+      PRIVATE_CASE_TYPE_OUTPLACE(SCALARTYPE1, sycl::ext::oneapi::bfloat16, __VA_ARGS__) \
+      PRIVATE_CASE_TYPE_OUTPLACE(SCALARTYPE2, sycl::half, __VA_ARGS__)                  \
```

Comment:
this is wrong logic. SCALARTYPE1 and 2 can be anything.

you can just rename the macro to `SYCL_DISPATCH_FLOATING_TYPES` or SYCL_DISPATCH_REDUCED_FLOATING_TYPES